thedmstoolkit/character_generation.py

```diff
@@ -0,0 +1,557 @@
+            if x not in Character_Generator.RACE_DICT["All"]:
+                raise RuntimeError("Invalid List")
+        Race: str = race_list[
+            Character_Generator.Generators[GeneratorKey](0, len(race_list) - 1)
+        ]
+        return Race
+
+    @staticmethod
+    def generate_class(class_list: list[str], GeneratorKey: str) -> str:
+        """
+        Returns a random class from the provided list
+        Args:
+            class_list (List): The list chosen from the Class dictionary
+            GeneratorKey (str): key required to specify which randomizer you want
+
+        Raises:
+            RuntimeError: If the key provided is the wrong type
+            RuntimeError: If there is an unauthorized item in the passed list
+
+        Returns:
+            String: the random class
+        """
+        if GeneratorKey not in Character_Generator.Generators:
+            raise RuntimeError("Generator does not exist")
+        for x in class_list:
+            if x not in Character_Generator.CLASS_DICT["All"]:
+                raise RuntimeError("Invalid List")
+        clazz: str = class_list[
+            Character_Generator.Generators[GeneratorKey](0, len(class_list) - 1)
+        ]
+        return clazz
+
+    @staticmethod
+    def generate_alignment(alignment_list: list[str], GeneratorKey: str) -> str:
+        """
+            Returns a random alignment from the provided list
+        Args:
+            AlignmentList (List): List of alignments from alignment dictionary
+            GeneratorKey (str): key required to specify which randomizer you want
+
+        Raises:
+            RuntimeError: If the key provided is the wrong type
+            RuntimeError: If there is an unauthorized item in the passed list
+
+        Returns:
+            String: Alignment
+        """
+        if GeneratorKey not in Character_Generator.Generators:
+            raise RuntimeError("Generator does not exist")
+        for x in alignment_list:
+            if x not in Character_Generator.ALIGNMENT_DICT["All"]:
+                raise RuntimeError("Invalid List")
+        alignment = alignment_list[
+            Character_Generator.Generators[GeneratorKey](0, len(alignment_list) - 1)
+        ]
+        return alignment
+
+    @staticmethod
+    def generate_background(background_list: list[str], GeneratorKey: str) -> str:
+        """
+            Generate a random background
+        Args:
+            background_list (str): background
+            GeneratorKey (str): key required to specify which randomizer you want
+
+        Raises:
+            RuntimeError: If the key provided is the wrong type
+            RuntimeError: If there is an unauthorized item in the passed list
+
+        Returns:
+            String: Background
+        """
+        if GeneratorKey not in Character_Generator.Generators:
+            raise RuntimeError("Generator does not exist")
+        if background_list != Character_Generator.BACKGROUND_LIST:
```

Comment:
If we require the list to be the one provided in the class, we should just make this `self.BACKGROUND_LIST` then we don't have to do this check.

Reply:
That's a leftover of the old check, it's now been updated